### PR TITLE
docs: clarify stacks' graduation process

### PR DESCRIPTION
**Pull request by skoh7645**
## Checklist

- [x] commit message follows [commit guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).
- [x] website builds and runs in production - for information on how to test locally go [here](https://github.com/appsody/website/blob/master/DEVELOPMENT.md).

## Description
Clarify the graduation process for stacks

## Related Issues
Fixes: #199

### CODE REVIEW

File: content/docs/stacks/stacks-overview.md [hunk 1]
```diff
@@ -51,15 +51,45 @@ The `templates` directory contains one or more starter applications that are cre
...
-## Stack categories
-
-Stacks are categorized as `stable`, `incubator` or `experimental` depending on the content of the stack.
-
-- `stable`: Stable stacks meet this set of [technical requirements](https://github.com/appsody/stacks/blob/master/TECHNICAL_REQUIREMENTS.md).
-
-- `incubator`: The stacks in the incubator folder are actively being worked on to satisfy the stable criteria.
-
-- `experimental`: Experimental stacks are not being actively worked on and might not fulfill the requirements of an Appsody stack. These can be used for trying out specific capabilities or proof of concept work.
+## Stack Stability Levels
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: To fit with other titles, please change to **Stack stability levels**

I think the first para needs to introduce the fact that there are 3 repositories that correspond to each stack stability level: experimental, incubator, and stable. There is nothing in the preceding text or topics to explain this. 

I think the information about how to contribute a stack should come after the explanations for each stack, not upfront as this detracts from the explanations for each category. i.e tell them what the criteria is first, then say ... if you plan to contribute a stack, you must follow the instructions in the guide to develop, test etc, and based on your results, determine which set of criteria your stack meets. Then raise a PR .... etc

File: content/docs/stacks/stacks-overview.md [hunk 2]
```diff
@@ -51,33 +51,26 @@ The `templates` directory contains one or more starter applications that are cre
...
 
-These same steps will occur for subsequent pull requests updating your stack. If you would like to move the stack to a different repository, you will need to submit a pull request, changing the stack from the current repository to the new repository directory. We will make a decision with regards to whether or not your stack fulfils the criteria of the new repository.
-
-The stability criteria for each Appsody repository is as follows:
-
-### Experimental - Not production-ready, proof of concept work that should be considered unstable and may be subject to breaking changes
-The stack should:
+### Experimental
+These stacks are not production-ready and should be considered proof of concept work that are unstable, and subject to breaking changes. They should:
 - Strictly adhere to the Appsody stack structure (i.e. [appsody stack lint](https://appsody.dev/docs/using-appsody/cli-commands#appsody-stack-lint) passes without errors)
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: Strictly adhere to the Appsody stack structure - the "strictly" wording sounds strange given these are experimental stack, can't we just say Adhere to the Appsody stack structure?

File: content/docs/stacks/stacks-overview.md [hunk 3]
```diff
@@ -51,15 +51,38 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Provide documentation with regards to the limitations of the stack in the `README.md`
+- Maintainers must remain active in terms of contributions and reviews
+- Support the [appsody test](https://appsody.dev/docs/using-appsody/cli-commands#appsody-test) and [appsody deploy](https://appsody.dev/docs/using-appsody/cli-commands#appsody-deploy) commands
+
+### Stable
+These stacks are production-ready. They should encompass the incubator criteria and further:
+- Support all Appsody CLI commands
+- Possess no open, critical issues
+- Pass on [appsody stack validate](https://appsody.dev/docs/using-appsody/cli-commands#appsody-stack-validate) and integration tests, on all 3 OS supported by Appsody without errors
+- Specify minimum requirements required by the stack, in the `stack.yaml` (i.e. the minimum Appsody/Docker/Buildah version required)
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: requirements required? Can't this be simplified to something like "Specify the minimum Appsody, Docker, or Buildah versions required in the `stack.yaml`."

File: content/docs/stacks/stacks-overview.md [hunk 4]
```diff
@@ -72,3 +95,12 @@ Below are the URLs for official Appsody repository releases.
 | `experimental` | `https://github.com/appsody/stacks/releases/latest/download/experimental-index.yaml` |
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing a stack
+
+If you would like to contribute a stack, you would need to submit a pull request and decide which repository you think is most suitable, corresponding to the stability level of your stack. We will review it, provide further guidance and request changes as necessary. If we agree that the new stack fulfils the criteria of your chosen repository, we will approve, merge and release your stack. When we release your stack, the following will occur:
+- A template archive for each stack template will be generated
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: Remove future tense by changing this to:
"When the stack is released:
 - A template archive is generated for each stack template
 - A docker image of the stack is pushed to [Appsody’s DockerHub](https://hub.docker.com/u/appsody)
 - The indexes for each repository are updated"

File: content/docs/stacks/stacks-overview.md [hunk 5]
```diff
@@ -51,15 +51,38 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Support all Appsody CLI commands
+- Possess no open, critical issues
+- Pass on [appsody stack validate](https://appsody.dev/docs/using-appsody/cli-commands#appsody-stack-validate) and integration tests, on all 3 OS supported by Appsody without errors
+- Specify minimum requirements required by the stack, in the `stack.yaml` (i.e. the minimum Appsody/Docker/Buildah version required)
+- Support the [appsody build](https://appsody.dev/docs/using-appsody/cli-commands#appsody-build) command with Buildah
+- Prevent creation of files on the local system that cannot be removed (i.e. files owned by root or other users)
+- Specify explicit versions for all required Docker images
+- Tag the stack with a major version (i.e. at least 1.0.0)
+- Provide documentation with regards to the migration of existing projects to use the stack
+- Exemplify [Docker best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) with particular attention paid to:
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: I find "Exemplify" and "with particular attention paid to" a little off-putting, how about "Follow Docker best practices, including:
 - Minimize the size of production images
- Use official base images
- Ensure security updates are included <- I'm not sure what we mean by this, do we mean "Images must not have any major security vulnerabilities"?
- Containers must be run by non-root users

File: content/docs/stacks/stacks-overview.md [hunk 6]
```diff
@@ -51,15 +51,38 @@ The `templates` directory contains one or more starter applications that are cre
...
+    1. An effort has been made to minimise the size of production images 
+    2. Use the official base images and ensure security updates are brought into the final production image as necessary
+    3. Only create containers that are run by non-root users
+
+Ultimately, you will decide if the runtime is production-ready, and we will decide if the stack is production-ready for Appsody.
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: Please can we delete this sentence it's not use? A more details version of this information should be in contributing a stack, or merging and releasing a stack.

File: content/docs/stacks/stacks-overview.md [hunk 7]
```diff
@@ -51,15 +51,38 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Support the [appsody build](https://appsody.dev/docs/using-appsody/cli-commands#appsody-build) command with Buildah
+- Prevent creation of files on the local system that cannot be removed (i.e. files owned by root or other users)
+- Specify explicit versions for all required Docker images
+- Tag the stack with a major version (i.e. at least 1.0.0)
+- Provide documentation with regards to the migration of existing projects to use the stack
+- Exemplify [Docker best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) with particular attention paid to:
+    1. An effort has been made to minimise the size of production images 
+    2. Use the official base images and ensure security updates are brought into the final production image as necessary
+    3. Only create containers that are run by non-root users
+
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: To me I think that somewhere in here we should clearly spell out:
1. Must include a detailed README.md, including:
 - Short description of the stack
 - Any prerequisites, requirements, or limitations

File: content/docs/stacks/stacks-overview.md [hunk 8]
```diff
@@ -72,3 +95,12 @@ Below are the URLs for official Appsody repository releases.
 | `experimental` | `https://github.com/appsody/stacks/releases/latest/download/experimental-index.yaml` |
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing a stack
+
+If you would like to contribute a stack, you would need to submit a pull request and decide which repository you think is most suitable, corresponding to the stability level of your stack. We will review it, provide further guidance and request changes as necessary. If we agree that the new stack fulfils the criteria of your chosen repository, we will approve, merge and release your stack. When we release your stack, the following will occur:
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: "We will review it..." shouldn't we be talking in terms of an Appsody stack maintainer, or an Appsody committer not just "we"?

File: content/docs/stacks/stacks-overview.md [hunk 9]
```diff
@@ -72,3 +95,12 @@ Below are the URLs for official Appsody repository releases.
...
+- A template archive for each stack template will be generated
+- A docker image of the stack will be pushed to [Appsody’s DockerHub](https://hub.docker.com/u/appsody)
+- The indexes for each repository will be updated
+
+These same steps will occur for subsequent pull requests updating your stack. If you would like to move the stack to a different repository, you will need to submit a pull request, changing the stack from the current repository to the new repository directory. We will make a decision with regards to whether or not your stack fulfils the criteria of the new repository.
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: I'm not sure the first sentence of this paragraph is very helpful, I'd expect that to be the case, so in my opinion that could be removed.

The next bit would read better if it was written in terms of what you're trying to do... something like... "To move a stack from incubator to stable, or experimental to incubator, stack maintainers need to/must..."

File: content/docs/stacks/stacks-overview.md [hunk 10]
```diff
@@ -51,15 +51,38 @@ The `templates` directory contains one or more starter applications that are cre
...
-
-- `incubator`: The stacks in the incubator folder are actively being worked on to satisfy the stable criteria.
-
-- `experimental`: Experimental stacks are not being actively worked on and might not fulfill the requirements of an Appsody stack. These can be used for trying out specific capabilities or proof of concept work.
+## Stack stability levels
+
+The criteria for each Appsody stack stability level is as follows:
+
+### Experimental
+These stacks are not production-ready and should be considered proof of concept work that are unstable, and subject to breaking changes. They should:
```
Comment (helenmasters) [inline content/docs/stacks/stacks-overview.md]: that is unstable, not that are unstable.
Maybe "These stacks are not production-ready and are considered as proof of concept. They might be unstable, and subject to breaking changes."

File: content/docs/stacks/stacks-overview.md [hunk 11]
```diff
@@ -72,3 +98,12 @@ Below are the URLs for official Appsody repository releases.
 | `experimental` | `https://github.com/appsody/stacks/releases/latest/download/experimental-index.yaml` |
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing a stack
+
```
Comment (neeraj-laad) [inline content/docs/stacks/stacks-overview.md]: I would like to re-phrase this section with 3 types of users.. 

Contributor - any contributor can submit a PR for a stack change and must use the above criteria to help improve the stack.

Stack Maintainer(s) - core maintainers of the stack as listed in stack.yaml. These are the people who will review and approve PR for stack changes from contributors and maintainers. They must use the above criteria and proactively improve the stack and resolve any issues.

Stack Release Team - Team responsible or Appsody stack releases. They must ensure that one of the stack maintainers has approved the change / PR before they merge the PR and create a release for the stack.

Please feel free to suggest another name for "stack release team" :)

Comment (neeraj-laad) [reply content/docs/stacks/stacks-overview.md]: I would prefer to see this section laid out as a process with small steps and clearly articulate which role will perform what as we move a stack from a PR to a released stack.

File: content/docs/stacks/stacks-overview.md [hunk 12]
```diff
@@ -51,15 +51,41 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Support the [appsody test](https://appsody.dev/docs/using-appsody/cli-commands#appsody-test) and [appsody deploy](https://appsody.dev/docs/using-appsody/cli-commands#appsody-deploy) commands
+- Document the limitations of the stack in the `README.md`
+
+### Stable
+These stacks are production-ready. They should encompass the incubator criteria and further:
+- Support all Appsody CLI commands
+- Possess no open, critical issues
+- Pass on [appsody stack validate](https://appsody.dev/docs/using-appsody/cli-commands#appsody-stack-validate) and integration tests, on all 3 OS supported by Appsody without errors
+- Specify the minimum Appsody, Docker and Buildah versions required in the `stack.yaml`
+- Support the [appsody build](https://appsody.dev/docs/using-appsody/cli-commands#appsody-build) command with Buildah
```
Comment (neeraj-laad) [inline content/docs/stacks/stacks-overview.md]: Can we check if the build with `buildah` is covered by our integration tests?

File: content/docs/overview.md [hunk 13]
```diff
@@ -18,7 +18,7 @@ These are configurable technology stacks built with popular runtimes and framewo
 information, see [Appsody Stacks](/docs/stacks/stacks-overview).
 
 ### Appsody Hub
-The Hub is the central point of control for Appsody Stacks where you can find available stacks, create new stacks, or modify existing ones. You can use the Hub content in the public repo or clone it to provide a private Hub that's based on your requirements. By making changes to the Stacks in the Hub, you can deploy updates to any application that's been built on them, simply by restarting the application.
+The Hub is the central point of control for Appsody Stacks where you can find available stacks, create new stacks, or modify existing ones. There are 3 [stack stability levels](http://localhost:8000/docs/stacks/stacks-overview#stack-stability-levels) - stable, incubator, and experimental. Stable stacks are production-ready. You can use the Hub content in the public repo or clone it to provide a private Hub that's based on your requirements. By making changes to the Stacks in the Hub, you can deploy updates to any application that's been built on them, simply by restarting the application.
```
Comment (kilnerm) [inline content/docs/overview.md]: People also have the option of creating individual stacks, perhaps we should mention that here.

File: content/docs/stacks/stacks-overview.md [hunk 14]
```diff
@@ -51,15 +51,41 @@ The `templates` directory contains one or more starter applications that are cre
...
+
+### Incubator
+These stacks are not production-ready and require further development to satisfy the stable criteria. They should encompass the experimental criteria and further:
+- Maintainers must remain active in terms of contributions and reviews
+- Support the [appsody test](https://appsody.dev/docs/using-appsody/cli-commands#appsody-test) and [appsody deploy](https://appsody.dev/docs/using-appsody/cli-commands#appsody-deploy) commands
+- Document the limitations of the stack in the `README.md`
+
+### Stable
+These stacks are production-ready. They should encompass the incubator criteria and further:
+- Support all Appsody CLI commands
```
Comment (kilnerm) [inline content/docs/stacks/stacks-overview.md]: Can we be explicit on what these command's are? Or link out to some document that is a comprehensive list. (Perhaps we could have tests to evaluate this criteria?)

File: content/docs/stacks/stacks-overview.md [hunk 15]
```diff
@@ -51,15 +51,41 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Possess no open, critical issues
```
Comment (kilnerm) [inline content/docs/stacks/stacks-overview.md]: I think this state is to transitory to call out here, it is entirely possible a stack can have such an issue raised following its promotion to stable and I assume we are not going to relegate it.

File: content/docs/stacks/stacks-overview.md (hunk 12 above)
Comment (skoh7645) [reply content/docs/stacks/stacks-overview.md]: Currently it is not

File: content/docs/stacks/stacks-overview.md (hunk 14 above)
Comment (skoh7645) [reply content/docs/stacks/stacks-overview.md]: We could link to the CLI Reference page on the website, other than that we don't have any tests that validate all commands

File: content/docs/stacks/stacks-overview.md (hunk 15 above)
Comment (skoh7645) [reply content/docs/stacks/stacks-overview.md]: Makes sense, I will remove this

File: content/docs/stacks/stacks-overview.md [hunk 16]
```diff
@@ -72,3 +97,21 @@ Below are the URLs for official Appsody repository releases.
 | `experimental` | `https://github.com/appsody/stacks/releases/latest/download/experimental-index.yaml` |
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing a stack
+Any contributor can submit a pull request to change a stack. However, typically, the Stack Release Team will wait for the Stack Maintainers' (as listed in the `stack.yaml`) approval, before approving and merging a stack change.
```
Comment (neeraj-laad) [inline content/docs/stacks/stacks-overview.md]: Typically, the Stack Release Team will wait for the Stack Maintainers (as listed in the `stack.yaml`) to review and approve the change, before merging and releasing stack change.

File: content/docs/stacks/stacks-overview.md [hunk 17]
```diff
@@ -72,3 +97,20 @@ Below are the URLs for official Appsody repository releases.
...
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing to a stack
+
+The following outlines the process to contributing changes to a stack:
+
+1. The Stack Contributor submits a pull request with changes to a stack. The pull request should follow commit guidelines detailing how the stack changes satisfy the stability criteria (e.g. how Docker best practices have been implemented). 
+
+    > Any contributor can submit a pull request to promote a stack to a higher stability level.
```
Comment (neeraj-laad) [inline content/docs/stacks/stacks-overview.md]: I'd avoid writing this explicitly

File: content/docs/stacks/stacks-overview.md [hunk 18]
```diff
@@ -72,3 +97,20 @@ Below are the URLs for official Appsody repository releases.
 | `experimental` | `https://github.com/appsody/stacks/releases/latest/download/experimental-index.yaml` |
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing to a stack
+
+The following outlines the process to contributing changes to a stack:
+
+1. The Stack Contributor submits a pull request with changes to a stack. The pull request should follow commit guidelines detailing how the stack changes satisfy the stability criteria (e.g. how Docker best practices have been implemented).
```
Comment (neeraj-laad) [inline content/docs/stacks/stacks-overview.md]: The pull request should follow commit guidelines detailing the stack changes. If the change is updating stack's stability level, it must also describe how the stability criteria is met (e.g. how Docker best practices have been implemented).

File: content/docs/stacks/stacks-overview.md [hunk 19]
```diff
@@ -72,3 +97,20 @@ Below are the URLs for official Appsody repository releases.
...
+
+## Contributing to a stack
+
+The following outlines the process to contributing changes to a stack:
+
+1. The Stack Contributor submits a pull request with changes to a stack. The pull request should follow commit guidelines detailing how the stack changes satisfy the stability criteria (e.g. how Docker best practices have been implemented). 
+
+    > Any contributor can submit a pull request to promote a stack to a higher stability level.
+
+2. The Stack Maintainer(s) (as listed in the `stack.yaml`) reviews the pull request, keeping in mind the stability criteria of the stack. In the case of the contribution of a new stack, the Stack Release Team reviews the pull request.
```
Comment (neeraj-laad) [inline content/docs/stacks/stacks-overview.md]: review

File: content/docs/stacks/stacks-overview.md [hunk 20]
```diff
@@ -72,3 +97,20 @@ Below are the URLs for official Appsody repository releases.
...
+
+3. If the Stack Maintainer(s) are satisfied the stack changes follow the stability criteria, the pull request gets approved. In the case of the contribution of a new stack, the Stack Release Team decides if the stack satisfies the chosen stability criteria.
```
Comment (neeraj-laad) [inline content/docs/stacks/stacks-overview.md]: Perhaps 2 and 3 could be merged as a review and approve step.

File: content/docs/stacks/stacks-overview.md (hunk 14 above)
Comment (kilnerm) [reply content/docs/stacks/stacks-overview.md]: A link to the CLI reference would be useful. Perhaps we could raise an issue to look at adding some form of acceptance tests?

File: content/docs/stacks/stacks-overview.md [hunk 21]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
-Stacks are categorized as `stable`, `incubator` or `experimental` depending on the content of the stack.
-
-- `stable`: Stable stacks meet this set of [technical requirements](https://github.com/appsody/stacks/blob/master/TECHNICAL_REQUIREMENTS.md).
-
-- `incubator`: The stacks in the incubator folder are actively being worked on to satisfy the stable criteria.
-
-- `experimental`: Experimental stacks are not being actively worked on and might not fulfill the requirements of an Appsody stack. These can be used for trying out specific capabilities or proof of concept work.
+## Stack stability levels
+
+The criteria for each Appsody stack stability level is as follows:
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Suggest a lead in sentence at the beginning and an explanation of what follows (sections) for clarity:

"Appsody stacks belong to one of three categories, depending on their stability level. The criteria for each category is detailed in the following sections:"

File: content/docs/stacks/stacks-overview.md [hunk 22]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
+
+### Experimental
+These stacks are not production-ready and are considered as proof of concept. They might be unstable, and subject to breaking changes. They should:
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Replace "They should" with

"Experimental stacks must meet the following criteria:"

File: content/docs/stacks/stacks-overview.md [hunk 23]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Adhere to the Appsody stack structure (i.e. [appsody stack lint](/docs/using-appsody/cli-commands#appsody-stack-lint) passes without errors)
+- Support the [appsody init](/docs/using-appsody/cli-commands#appsody-init), [appsody run](/docs/using-appsody/cli-commands#appsody-run) and [appsody build](/docs/using-appsody/cli-commands#appsody-build) commands
+
+### Incubator
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Replace "They should..."

"Incubator stacks must meet all the criteria set for experimental stacks and the following additional items:"

File: content/docs/stacks/stacks-overview.md [hunk 24]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
-
-- `experimental`: Experimental stacks are not being actively worked on and might not fulfill the requirements of an Appsody stack. These can be used for trying out specific capabilities or proof of concept work.
+## Stack stability levels
+
+The criteria for each Appsody stack stability level is as follows:
+
+### Experimental
+These stacks are not production-ready and are considered as proof of concept. They might be unstable, and subject to breaking changes. They should:
+- Adhere to the Appsody stack structure (i.e. [appsody stack lint](/docs/using-appsody/cli-commands#appsody-stack-lint) passes without errors)
+- Support the [appsody init](/docs/using-appsody/cli-commands#appsody-init), [appsody run](/docs/using-appsody/cli-commands#appsody-run) and [appsody build](/docs/using-appsody/cli-commands#appsody-build) commands
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Please add a "comma" after (/docs/using-appsody/cli-commands#appsody-run).

File: content/docs/stacks/stacks-overview.md [hunk 25]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
+
+### Incubator
+These stacks are not production-ready and require further development to satisfy the stable criteria. They should encompass the experimental criteria and further:
+- Maintainers must remain active in terms of contributions and reviews
+- Support the [appsody test](/docs/using-appsody/cli-commands#appsody-test) and [appsody deploy](/docs/using-appsody/cli-commands#appsody-deploy) commands
+- Document the limitations of the stack in the `README.md`
+
+### Stable
+These stacks are production-ready. They should encompass the incubator criteria and further:
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Similarly, please rephrase "They should ..."

"Stable stacks must meet all the criteria set for experimental and incubator stacks, and the following additional items:"

File: content/docs/stacks/stacks-overview.md [hunk 26]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Support all [Appsody CLI commands](/docs/using-appsody/cli-commands)
+- Pass on [appsody stack validate](/docs/using-appsody/cli-commands#appsody-stack-validate) and integration tests, on all 3 OS supported by Appsody without errors
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Please expand OS -> operating systems

File: content/docs/stacks/stacks-overview.md [hunk 27]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Specify the minimum Appsody, Docker and Buildah versions required in the `stack.yaml`
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Oxford comma after "Docker" please

File: content/docs/stacks/stacks-overview.md [hunk 28]
```diff
@@ -51,15 +51,40 @@ The `templates` directory contains one or more starter applications that are cre
...
+- Support the [appsody build](/docs/using-appsody/cli-commands#appsody-build) command with Buildah
+- Prevent creation of files on the local system that cannot be removed (i.e. files owned by root or other users)
+- Specify explicit versions for all required Docker images
+- Tag the stack with a major version (i.e. at least 1.0.0)
+- Provide documentation with regards to the migration of existing projects to use the stack
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: Provide migration documentation for any existing projects that use the stack when changes are made.

(Assume this is what is meant?)

File: content/docs/stacks/stacks-overview.md [hunk 29]
```diff
@@ -72,3 +97,18 @@ Below are the URLs for official Appsody repository releases.
 | `experimental` | `https://github.com/appsody/stacks/releases/latest/download/experimental-index.yaml` |
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing to a stack
+
+The following outlines the process to contributing changes to a stack:
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: The following steps outline the process for contributing changes to a stack:

File: content/docs/stacks/stacks-overview.md [hunk 30]
```diff
@@ -72,3 +97,18 @@ Below are the URLs for official Appsody repository releases.
...
+
+1. The Stack Contributor submits a pull request with changes to a stack. The pull request should follow commit guidelines detailing the stack changes. If the change is updating the stack's stability level, it must also describe how the stability criteria is met (e.g. how Docker best practices have been implemented). 
+
+2. The Stack Maintainer(s) (as listed in the `stack.yaml`) review the pull request, keeping in mind the stability criteria of the stack. If the Stack Maintainer(s) are satisfied that the stack changes follow the stability criteria, the pull request gets approved. 
+
+    >Note: In the case of a contribution of a new stack, the Stack Release Team reviews the pull request and decides if the stack satisfies the chosen stability criteria.
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: In the case of a contribution of a new stack -> If the stack is new, .....

File: content/docs/stacks/stacks-overview.md [hunk 31]
```diff
@@ -72,3 +97,18 @@ Below are the URLs for official Appsody repository releases.
 | `experimental` | `https://github.com/appsody/stacks/releases/latest/download/experimental-index.yaml` |
 
 By default, Appsody comes with the `incubator` and `experimental` repositories. Other repositories can be added by running the [`appsody repo add`](/docs/using-appsody/cli-commands/#appsody-repo-add) command.
+
+## Contributing to a stack
+
+The following outlines the process to contributing changes to a stack:
+
+1. The Stack Contributor submits a pull request with changes to a stack. The pull request should follow commit guidelines detailing the stack changes. If the change is updating the stack's stability level, it must also describe how the stability criteria is met (e.g. how Docker best practices have been implemented).
```
Comment (SueChaplain) [inline content/docs/stacks/stacks-overview.md]: If the change is updating the stack's stability level, it must also describe
->
If the change improves the stack stability level, the pull request must also describe how the new criteria for that level is met ....

File: content/docs/stacks/stacks-overview.md (hunk 29 above)
Comment (SueChaplain) [reply content/docs/stacks/stacks-overview.md]: Actually should this say "The following steps outline the process for contributing a new stack or changes to a stack:" as we have a note further down that basically says "if this is a new stack..."?

File: content/docs/stacks/stacks-overview.md (hunk 28 above)
Comment (skoh7645) [reply content/docs/stacks/stacks-overview.md]: It meant migrate user's existing applications (without Appsody) to use the Appsody stack

Comment (SueChaplain) [reply content/docs/stacks/stacks-overview.md]: Provide documentation to help users with existing projects migrate to using the stack?

File: content/docs/stacks/stacks-overview.md (hunk 14 above)
Comment (skoh7645) [reply content/docs/stacks/stacks-overview.md]: @kilnerm https://github.com/appsody/appsody/issues/860 feel free to add more context to it :)

**Comment by neeraj-laad** [comment]
Looks good as a first draft. I believe the criteria for each will need to evolve further, but this is a good first pass at it.

**Pull request merged**